File: uStock-Development-Android/app/src/main/java/data_structures/wallet_model/Transaction.kt
package data_structures.wallet_model
import data_structures.DateSerializer
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.util.Date

@Serializable()
data class Transaction(
    @SerialName("_id") var id: String,
    val sender: String,
    val recipient: String,
    val amount: Double,
    @SerialName("stockPercentagePurchased") val stockPercentagePurchased: Double?,
    @Serializable(with = DateSerializer::class) val date: Date,
    val type: String,
    val stock: String
)

File: uStock-Development-Android/taskThree_models/Data Structures/WalletModels/Transaction.swift
//
//  Transaction.swift
//  uStockIOS-V2-AI
//
//  Created by Gal Cohavy on 6/7/23.
//

import Foundation

public struct Transaction: Codable, Identifiable {
    public let id: String
    let sender: String
    let recipient: String
    let amount: Double
    let stockPercentagePurchased: Double?
    let date: Date
    let type: String
    let stock: String
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case sender
        case recipient
        case amount
        case stockPercentagePurchased
        case date
        case type
        case stock
    }
}
